graphql/schemabuilder/pagination.go

```diff
@@ -766,19 +796,57 @@ func (c *connectionContext) consumeSorts(sb *schemaBuilder, m *method, typ refle
 	c.SortFunctions = make(map[string]func([]sortReference, SortOrder))
 	c.SortFields = make(map[string]*graphql.Field)
 
-	for name, fn := range m.SortFuncs {
-		sortableTyp := getFuncReturnType(fn)
-		if !supportedSort(sortableTyp) {
-			return fmt.Errorf(
-				"invalid sort field %s: unsupported return type %v, must be of kind int, uint, float or string",
-				name,
-				sortableTyp,
-			)
+	for name, fn := range m.SortFuncsStruct {
+		if fn.SortFunc != nil {
```

Comment:
can we put all of this into a `getFieldForSortStruct` function?  This for loop is pretty complicated. (Possible for follow-up)